docs/coding-guidelines.md

```diff
@@ -429,9 +425,83 @@ for the following situations. Note that these changes should still be recorded a
+  users, this is the time to update them\!  
+* Provide users with tools to understand the breaking change  
+  * (Optional) Create or update a **Github issue** to document what the change is about, who it
+    affects and what its effects are  
+  * (Optional) Consider adding **telemetry** that allows users to track their migration. For
+    example, you can add a counter for the times that you see a payload that would be affected by
+    the breaking change.  
+* Notify users about the upcoming change  
+  * Add a **changelog entry** that describes the feature gate. It should include its name, when you
+    may want to use it, and what its effects are. The changelog entry can be given the `enhancement`
+    classification at this stage.
+  * (Optional but strongly recommended) Log a **warning** if the user is using the software in a way
+    that would be affected by the breaking change. Point the user to the feature gate and any
+    official docs.  
+* (Optional) Try to **test this in a realistic setting.** If this solves an issue, ask the poster to
+  try to use it and check that everything works.
+
+##### Beta stage
+
+At the beta stage, the change is opt-out. At this stage we want to notify users that the change is
+happening, and help them understand how to revert back to the previous behavior temporarily if they
+need to do so. You may directly start from this stage for breaking changes that are less impactful
+or for changes that should not have a functional impact such as performance changes. Consider the
+following items before moving from alpha to beta:
+
+* Schedule the **docs and examples** update to align with the breaking change release if you
+  couldn’t do it before  
+* Provide users with tools to understand the breaking change  
+  * Update the **Github issue** with the new default behavior (or create one if starting from here)  
+  * Update the feature gate to add the ‘to version’ to the feature gate  
+* Notify users about the change  
+  * Add a second **changelog entry** that describes the change one more time and is marked as
+    ‘breaking’.  
+  * If applicable, add an **error message** that tells you this is the result of a breaking change
+    that can be temporarily reverted disabling the feature gate and points to any issue or docs
+    about it.
+
+##### Stable stage
+
+At the stable stage, the change cannot be reverted. In some cases, you may directly start here and
+just do the change, in which case you do not need a feature gate, but you should still follow the
+checklist below (notify, update docs and examples). Consider the following items before moving from
+beta to stable:
+
+* Remove the dead code  
+* Provide users with tools to understand the breaking change  
+  * Update the **documentation** **and examples** to remove any references to the feature gate and
+    the previous behavior. Close the **Github issue** if you opened one before.  
+* Notify users about the change  
```

Comment:
```suggestion
* Remove the dead code
* Provide users with tools to understand the breaking change
  * Update the **documentation** **and examples** to remove any references to the feature gate and
    the previous behavior. Close the **Github issue** if you opened one before.
* Notify users about the change
```
You can see in the negative part of this diff the spaces that were removed. Not super important, just something I noticed.

Reply:
Sorry, not sure how I missed it, I fixed it now